drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -1960,8 +1965,28 @@ void RasterizerSceneGLES3::render_scene(const Ref<RenderSceneBuffers> &p_render_
 	}
 	RENDER_TIMESTAMP("Render Opaque Pass");
 	uint64_t spec_constant_base_flags = 0;
+	bool disable_linear_to_srgb = false;
 
 	{
+		// Check if our target is an SRGB target, if we're on OpenGL ES or if sRGB conversion is enabled,
+		// the hardware expects our fragment shader to output in linear color space
+		// and will perform the conversion for us.
+		GLint encoding = GL_LINEAR;
+		glGetFramebufferAttachmentParameteriv(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_FRAMEBUFFER_ATTACHMENT_COLOR_ENCODING, &encoding);
```

Comment:
We may want to store this on the framebuffer we're using and may need to check if there are devices that don't support this check. As far as I can tell this was made core in OpenGL 3.0 and GLES3.